add --cert-name and --deploy-hook options

Hoeze opened the pull request:
Can be used like this:
```yaml
certbot_certs:
  - name: squid
    domains:
      - "{{ ansible_hostname }}.{{ domain_zone }}"
    deploy_hook: >-
      install -o squid -g squid -m 0600 $RENEWED_LINEAGE/fullchain.pem /etc/squid/fullchain.pem;
      install -o squid -g squid -m 0600 $RENEWED_LINEAGE/privkey.pem /etc/squid/privkey.pem;
      systemctl reload squid || true
```

Hoeze commented:
Does someone know why this fails? Seems unrelated to my PR:
```
  fatal: [instance]: FAILED! => {"ansible_facts": {}, "changed": false, "failed_modules": {"ansible.legacy.setup": {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python3.9"}, "failed": true, "module_stderr": "sudo: PAM account management error: Authentication service cannot retrieve authentication info\nsudo: a password is required\n", "module_stdout": "", "msg": "MODULE FAILURE: No start of json char found\nSee stdout/stderr for the exact error", "rc": 1}}, "msg": "The following modules failed to execute: ansible.legacy.setup\n"}
```

geerlingguy commented:
How do these changes interact with the updates from #208 from @theS1LV3R ? Sorry don't have time to do a full review at the current moment :)

Hoeze commented:
- #208 relies on the certbot command output to determine whether running it changed anything.
  This could be problematic. At least in my case it led to requesting a new certificate each time, exhausting my request limit.
- This PR checks first whether an update is necessary at all by comparing the list of domain names. This way it is also possible to **remove** domains (I tested that, it works). As far as I can tell, `--expand` only **adds** additional domains.

So at least for me, this PR makes `--expand` obsolete.
@geerlingguy I hope this helps.

geerlingguy commented:
Is it possible to rebase this PR on master? I don't see the changes from https://github.com/geerlingguy/ansible-role-certbot/pull/208/files being updated, it looks like the branch forks from somewhere prior to current master.

Hoeze commented:
@geerlingguy sorry for the delay, I missed your answer.
I have merged the master branch now.

The reason why I started from #223 instead of #208 is that after some testing I noticed that `--expand` does not do exactly what I wanted. My aim was to have an idempotent role which ensures that exactly and only those domains are in the certificate that are also listed in the role. With `--expand`, this was not the case when one added and removed again a domain.

Also, #208 checks if there was a change based on the output of the certbot command:
```yaml
changed_when: "'no action taken' not in certbot_create.stdout"
```
This PR provides a much more powerful approach as it explicitly checks the domains that a certain (named) certificate has.
```yaml
- name: Determine if domains have changed
  set_fact:
    letsencrypt_cert_domains_changed: "{{ letsencrypt_cert_domains != (cert_item.domains | map('trim') | select('!=', '') | list | sort) }}"
```

TL;DR: `--expand` is still available, but useless when the role explicitly checks the desired domains per certificate.

Hoeze commented:
@geerlingguy if you wish I can squash the commit for a clean history.

geerlingguy commented on .github/workflows/ci.yml:
```diff
@@ -44,10 +44,10 @@ jobs:
             playbook: converge.yml
             experimental: false
-          - distro: ubuntu2404
+          - distro: ubuntu2204
```
This is reverting testing back to older releases of Ubuntu and Debian FYI - please make sure these are both kept at 24.04 and 12.

Hoeze replied:
:+1: Will fix. Would you accept a PR on testing multiple Ubuntu versions?
It would be nice to know whether the PR works on older releases as well :)

Hoeze commented:
Hi @geerlingguy, would you have time to check this PR again?
- Fixed the distribution versions tested
- Removed duplicate 'certbot_create_extra_args' introduced in #227

Hoeze commented:
Thx @geerlingguy!
